hathor/p2p/sync_v2/manager.py

```diff
@@ -0,0 +1,1207 @@
+        self.log.debug('handle_get_block_txs', **data)
+        child_hash = bytes.fromhex(data['child'])
+        last_block_hash = bytes.fromhex(data['last_block'])
+        self.send_block_txs(child_hash, last_block_hash)
+
+    def send_block_txs(self, child_hash: bytes, last_block_hash: bytes) -> None:
+        try:
+            tx = self.tx_storage.get_transaction(child_hash)
+        except TransactionDoesNotExist:
+            # In case the tx does not exist we send a NOT-FOUND message
+            self.log.debug('requested child_hash not found', child_hash=child_hash.hex())
+            self.send_message(ProtocolMessages.NOT_FOUND, child_hash.hex())
+            return
+        if not self.partial_vertex_exists(last_block_hash):
+            # In case the tx does not exist we send a NOT-FOUND message
+            self.log.debug('requested last_block_hash not found', last_block_hash=last_block_hash.hex())
+            self.send_message(ProtocolMessages.NOT_FOUND, last_block_hash.hex())
+            return
+        if self.transactions_streaming is not None and self.transactions_streaming.is_running:
+            self.transactions_streaming.stop()
+        self.transactions_streaming = TransactionsStreaming(self, [tx], last_block_hash,
+                                                            limit=self.DEFAULT_STREAMING_LIMIT)
+        self.transactions_streaming.start()
+
+    # XXX/TODO: END DEPRECATED SECTION
+
+    def send_get_transactions_bfs(self, start_from: list[bytes], until_first_block: bytes) -> None:
+        """ Request a BFS of all transactions starting from start_from list and walking back into parents/inputs.
+
+        The start_from list can contain blocks, but they won't be sent. For example if a block B1 has T1 and T2 as
+        transaction parents, start_from=[B1] and start_from=[T1, T2] will have the same result.
+
+        The stop condition is reaching transactions/inputs that have a first_block of height less or equan than the
+        height of until_first_block. The other peer will return an empty response if it doesn't have any of the
+        stransactions in start_from or if it doesn't have the until_first_block block.
+        """
+        self._setup_tx_streaming()
+        start_from_hexlist = [tx.hex() for tx in start_from]
+        until_first_block_hex = until_first_block.hex()
+        self.log.debug('send_get_transactions_bfs', start_from=start_from_hexlist, last_block=until_first_block_hex)
+        payload = json.dumps(dict(
+            start_from=start_from_hexlist,
+            until_first_block=until_first_block_hex,
+        ))
+        self.send_message(ProtocolMessages.GET_TRANSACTIONS_BFS, payload)
+        self.receiving_stream = True
+
+    def handle_get_transactions_bfs(self, payload: str) -> None:
+        if self._is_streaming:
+            self.log.warn('ignore GET-TRANSACTIONS-BFS, already streaming')
+            return
+        data = json.loads(payload)
```

Comment:
Should we have some kind of protection in case the other fullnode tries to send something that is not a JSON? Or even perform a schema validation on the payload?

I'm thinking this in case someone tries to handcraft some bad payload to be sent.

Reply:
I feel we should just handle this error and close connection. In fact, the connection is already closed in case of an unhandled exception. So, handing the exception would just prevent this exception to generate alerts and incidents.

Reply:
I'll clean this up a little bit.